packages/react/src/elements/components/Icon/Icon.js

```diff
@@ -1,76 +1,124 @@
-/* eslint-disable consistent-return */
-import React, { Component } from "react";
+import React from "react";
 import PropTypes from "prop-types";
 import cx from "classnames";
+
 import "./icon.scss";
+import Icons, { names } from "../Icons/icons";
 
-import Icons from "../Icons/icons";
+const COMPONENT_CLASS = "hig__icon";
+const sizes = Object.freeze({ PX_24: "24", PX_16: "16" });
+/** @type {string[]} */
+const availableSizes = Object.values(sizes);
 
-const AvailableSizes = ["24", "16"];
+/**
+ * @typedef {Object} IconProps
+ * @property {string} name
+ * @property {string} svg
+ * @property {string} size
+ * @property {string} [nameOrSVG]
+ */
 
-export default class Icon extends Component {
-  setSizedIcon(icon, size) {
-    if (!(icon && size)) {
-      return; // Silently return until name and size are set
-    }
+/**
+ * @param {string} size
+ * @returns {boolean}
+ */
+function isValidSize(size) {
+  return availableSizes.includes(size);
+}
 
-    return AvailableSizes.indexOf(size) > -1
-      ? this._confirmNameOrSVG(icon, size)
-      : console.error(
-          `Icon named "${icon} size "${size}" not found, only these size are allowed: `,
-          AvailableSizes
-        );
-  }
+/**
+ * @param {string} name
+ * @param {string} size
+ * @returns {string} SVG markup
+ */
+function getIconMarkupFromName(name, size) {
+  const key = `${name}-${size}`;
+
+  return Icons[key];
+}
```

Comment:
👌 so this would return `undefined` if given an icon and a size that doesn't have a match.

Reply:
:+1

Reply:
@eskfung 

Correct, the goal is move and handle all of the existence checks in one place.